.gitignore

```diff
@@ -12,6 +11,20 @@ conf.pri
 Makefile*
 *.pyc
 *.log
+#==============================================================================#
+# Directories to ignore (do not add trailing '/'s, they skip symlinks).
+#==============================================================================#
+# VS2017 and VSCode config files.
+.vscode
+.vs
+
+.idea
+.*.sw?
+.sw?
+
+# clangd index. (".clangd" is a config file now, thus trailing slash)
+.clangd/
+.cache
```

Comment:
I hold the opinion that these entries should reside in your own .gitignore file, rather than every project you worked on.
See: https://git-scm.com/docs/git-config#Documentation/git-config.txt-coreexcludesFile

Reply:
Considering we are using Cmake as build system, it's OK with Cmake build directory?